libbpf-rs/src/object.rs

```diff
@@ -86,48 +155,65 @@ impl Map {
         unimplemented!();
     }
 
-    /// Same as [`Map::lookup()`] except this also deletes the key from the map.
+    /// Same as [`LoadedMap::lookup()`] except this also deletes the key from the map.
     ///
     /// `key` must have exactly [`Map::key_size()`] elements.
-    pub fn lookup_and_delete(&mut self, _key: &[u8], _opts: MapOptions) -> Result<Option<Vec<u8>>> {
+    pub fn lookup_and_delete(
+        &mut self,
+        _key: &[u8],
+        _flags: LoadedMapFlags,
```

Comment:
this is not loaded map flags, it's that operation's flags... Do you intend to make LoadedMapFlags into a builder? Wouldn't plain int be easier to use though?

Reply:
Yeah, let's use bitflags